### PR TITLE
Feature/validation result only

### PR DESCRIPTION
**Pull request by 5t33**
#### What does this PR do?
Added some functions to the validator which decouples the logging from the validation so that one can get only the validation results and use a logging function to log them.

### CODE REVIEW

**Comment by 5t33** [comment]
Auto-format is on so the diff might be a bit hard to parse.

File: src/main/java/tr/com/srdc/cda2fhir/validation/ValidatorImpl.java [hunk 1]
```diff
@@ -25,51 +25,76 @@
...
+		logger.info("Validating the bundle containing " + bundle.getEntry().size() + " entries");
+
+		List<ValidationResult> results = new ArrayList<ValidationResult>();
+
+		// traverse the entries of the bundle
+		for (BundleEntryComponent entry : bundle.getEntry()) {
+			if (entry != null && entry.getResource() != null) {
+				results.add(validateResourceResultOnly(entry.getResource()));
+			} else {
+				logger.warn("An entry of the bundle validator was running on was found null. Ignoring the entry.");
```
Comment (mmccall) [inline src/main/java/tr/com/srdc/cda2fhir/validation/ValidatorImpl.java]: can you just rephrase this error where it shows up? It's just bad english. "Bundle validator encountered a null entry, ignoring." would be better.

File: src/main/java/tr/com/srdc/cda2fhir/validation/ValidatorImpl.java [hunk 2]
```diff
@@ -105,97 +133,132 @@ public OutputStream validateBundle(Bundle bundle) {
 		} catch (IOException e) {
 			logger.error("Could not write to the output stream.", e);
 		}
-		
+
 		return outputStream;
 	}
-	
+
+	public void logValidationResults(List<ValidationResult> results) {
```
Comment (mmccall) [inline src/main/java/tr/com/srdc/cda2fhir/validation/ValidatorImpl.java]: I won't be using this logging function in Higgs, FYI. The results should be logged at the warn level over there.

File: src/main/java/tr/com/srdc/cda2fhir/validation/ValidatorImpl.java (hunk 1 above)
Comment (5t33) [reply src/main/java/tr/com/srdc/cda2fhir/validation/ValidatorImpl.java]: Will do. That's what was already in the validator when we got it.

File: src/main/java/tr/com/srdc/cda2fhir/validation/ValidatorImpl.java (hunk 2 above)
Comment (5t33) [reply src/main/java/tr/com/srdc/cda2fhir/validation/ValidatorImpl.java]: gotchya. I just removed the function.

**Review by mmccall — APPROVED** [review]
seems good, will try and use in higgs.
